specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,741 @@
+# Title
+
+**Author(s)**:  [Shruthi MN](https://github.com/Shruthi-1MN), [Pravin Ranjan](https://github.com/PravinRanjan10)
+
+## Summary
+
+A share is a remote, mountable file system. You can mount a share to and access a share from several hosts by several users at a time. we can create a share, list shares, and show information for, update, and delete a share.To create a share, specify one of these supported protocols:
+* NFS
+* CIFS
+
+## Motivation
+
+Multiple users can access the same file share mount point.
+Data that would otherwise be duplicated on each client can be kept in a single location and accessed by client on the network.
+
+### Goals
+
+* Facilitating File Share Service by providing Standard API to manage multiple vendors, simplify File Share API definition  
+* File share across the users based on access capability
+* Scope for now is to support only NFS, SMB file share protocols
+* write a Manila driver and a default NFS/SMB driver
+* Provide CLI support for File Share service
+
+### Non-Goals
+
+* Provide support for multiple other file share protocols ex: GlusterFS, HDFS etc.. 
+* Supporting Replication and Data Protection for file share profile
+
+## Design Details
+
+1.  A user can request for action (let’s say: create a share) through
+     Northbound Plugins(NBP). We have already few plugins, but need to
+     modify for file share request. A User can also use CLI or dashboard.
+     So, all the actions(create, list, update, delete) should be supported
+     by CLI and dashboard.
+2. Now, NBP can communicate to OpenSDS API server.
+3. Based on user request File Share API will be selected.
+4. Internally, OpenSDS File Share API talk to controllers.
+5. Controllers will also maintain the File Share metadata in database.
+6. Controller communicates to Dock, specific driver to create a share.
+
+![FileShare Architecture Diagram](FileShare_Design.png?raw=true "File_Share_Design")
+
+#### Flow diagram:
+![FileShare Architecture Diagram](FileShare_Flow_diagram.png?raw=true "File_Share_Design")
+
+### REST API impact
+
+YES
+Fileshare API's needs to write and Existing Profile API's needs to be changed.
+
+### Data model impact
+
+YES
+
+### Security impact
+
+No
+
+### Other end user impact
+
+No
+
+## Use Cases
+
+Multiple users can access the same file share mount point.
+Data that would otherwise be duplicated on each client can be kept in a single location and accessed by client on the network.
+
+## Implementation
+This is the list of proposal for API
+
+##### 1. GET /v1beta/{tenantId}/file/shares
+* Lists all shares.
+ ##### Request
+ ```json
+"tenantId": "string"
+```
+  ##### Response parameters
+```json
+  “id” : “string”
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
```

Comment:
Please make API specification dataType format consistent.

Please add sample API request/response as example.

Reply:
I think it's ok for design doc to have json format timing detail. It's more clear way to represent.